src/material_pool.hpp

```diff
@@ -37,130 +52,83 @@ namespace wr
 				 TextureHandle roughness,
 				 TextureHandle metallic,
 				 TextureHandle ao, 
-				 bool alpha_masked = false, 
-				 bool double_sided = false);
+				 float alpha_constant = false,
+				 float double_sided = false);
 
 		Material(const Material& rhs) = default;
 
 		~Material();
 
-		TextureHandle GetAlbedo() { return m_albedo; }
-		void SetAlbedo(TextureHandle albedo);
-
-		void UseAlbedoTexture(bool use_albedo);
-
-		TextureHandle GetNormal() { return m_normal; }
-		void SetNormal(TextureHandle normal);
+		TextureHandle GetTexture(MaterialTextureType type);
+		void SetTexture(MaterialTextureType type, TextureHandle handle);
+		void ClearTexture(MaterialTextureType type);
+		bool HasTexture(MaterialTextureType type);
 
-		void UseNormalTexture(bool use_normal);
+		template<uint16_t s>
+		void GetConstant(MaterialConstantType type, float(&val)[s]);
+		
+		template<uint16_t s>
+		void SetConstant(MaterialConstantType type, float(&val)[s]);
 
-		TextureHandle GetRoughness() { return m_rougness; }
-		void SetRoughness(TextureHandle roughness);
+		void SetConstant(MaterialConstantType type, float val);
+		void GetConstant(MaterialConstantType type, float &val);
 
-		void UseRoughnessTexture(bool use_roughness);
-
-		TextureHandle GetMetallic() { return m_metallic; }
-		void SetMetallic(TextureHandle metallic);
-
-		void UseMetallicTexture(bool use_metallic);
+		TexturePool* const GetTexturePool() { return m_texture_pool; }
 
-		TextureHandle GetAmbientOcclusion() { return m_ao; }
-		void SetAmbientOcclusion(TextureHandle ao);
+		ConstantBufferHandle* GetConstantBufferHandle() { return m_constant_buffer_handle; };
```

Comment:
Should have moved the impl to the cpp file and marked it const.

Reply:
It's just a getter, I agree with the const, but idk if having an implementation in the CPP for just a return is required.